templates/workspace_services/azureml/terraform/network.tf

```diff
@@ -0,0 +1,62 @@
+
+data "azurerm_network_security_group" "ws" {
+  name                = "nsg-ws"
+  resource_group_name = data.azurerm_resource_group.ws.name
+}
+
+
+resource "azurerm_network_security_rule" "allow-BatchNodeManagement-inbound-29877" {
+  access                      = "Allow"
+  destination_port_range      = "29877"
+  destination_address_prefix  = "VirtualNetwork"
+  source_address_prefix       = "BatchNodeManagement"
+  direction                   = "Inbound"
+  name                        = "BatchNodeManagement-inbound-29877"
+  network_security_group_name = data.azurerm_network_security_group.ws.name
+  priority                    = 200
+  protocol                    = "TCP"
+  resource_group_name         = data.azurerm_resource_group.ws.name
+  source_port_range           = "*"
+}
+
+resource "azurerm_network_security_rule" "allow-BatchNodeManagement-inbound-29876" {
+  access                      = "Allow"
+  destination_port_range      = "29876"
+  destination_address_prefix  = "VirtualNetwork"
+  source_address_prefix       = "BatchNodeManagement"
+  direction                   = "Inbound"
+  name                        = "BatchNodeManagement-inbound-29876"
+  network_security_group_name = data.azurerm_network_security_group.ws.name
+  priority                    = 201
+  protocol                    = "TCP"
+  resource_group_name         = data.azurerm_resource_group.ws.name
+  source_port_range           = "*"
+}
+
+resource "azurerm_network_security_rule" "allow-AzureMachineLearning-inbound-44224" {
+  access                      = "Allow"
+  destination_port_range      = "44224"
+  destination_address_prefix  = "VirtualNetwork"
+  source_address_prefix       = "*"
```

Comment:
Is there a tag so this can be narrowed down?

There is potential risk that a rogue service could be run on this port and a connection made (if can be routed - unlikely) from outside the VNet.

Reply:
Tried it: error indicated that this can be either "Internet" or "AzureMachineLearning". However when put to "AzureMachineLearning", the terminal didn't open.